### PR TITLE
Properly report the ID of newly added keys

### LINKED ISSUE #4656: add key does not report the new key's ID
Output of `restic version`
--------------------------
restic 0.16.3 compiled with go1.21.6 on linux/amd64

What backend/service did you use to store the repository?
---------------------------------------------------------
rest:http://

Problem description / Steps to reproduce
----------------------------------------
When adding a key to a repository, we currently have no "easy" way to know the new key's ID.
This is a problem because sometimes we need to pass this ID as a hint for other commands.

In particular, a key hint is needed when the repository has many keys - otherwise opening the repository may fail with "Fatal: maximum number of keys reached" even when a proper password is provided.

Expected behavior
-----------------

I suppose the author of the "key add" command wanted to have the ID printed like this:
```
saved new key as 116b1f6aaf7a299f6355bd7d892e2602e1a91f5b9ce8a69c6938ba3e4f1fcc79
```
Actual behavior
---------------

Currently `key add` displays instead:

```bash
restic -r /tmp/kevo/ key add
enter password for repository: 
repository 70789ef8 opened (version 2, compression level auto)
enter new password: 
enter password again: 
saved new key as <Key of xx@yyy, created on 2024-01-25 10:17:55.560224177 +0100 CET m=+10.381054308>
```
Do you have any idea what may have caused this?
-----------------------------------------------

The command dumps the Go object instead of the ID.
A trivial 1 line patch will be provided.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------

One day someone will outsmart all our defenses, burn our IT to the ground and destroy our other backups. Then the restic repos on the hard drives on the shelf it will save the day.

### PR DESCRIPTION
**Pull request by ls-initiatives**
What does this PR change? What problem does it solve?
-----------------------------------------------------

Other commands like key list and key remove show the key's ID.

Showing the ID here lets users easily reuse the ID as a key hint for subsequent commands. In particular, a key hint is needed when the repository has many keys - otherwise opening the repository may fail with "Fatal: maximum number of keys reached" even when a proper password is provided.

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------

Fixes #4656

- [X] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [ ] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added tests for all code changes.
- [x] I have added documentation for relevant changes (in the manual).
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [x] I have run `gofmt` on the code in all commits.
- [X] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [x] I'm done! This pull request is ready for review.

### CODE REVIEW

File: cmd/restic/cmd_key.go [hunk 1]
```diff
@@ -142,7 +142,7 @@ func addKey(ctx context.Context, repo *repository.Repository, gopts GlobalOption
 		return err
 	}
 
-	Verbosef("saved new key as %s\n", id)
+	Verbosef("saved new key as %s\n", id.ID())
```
Comment (MichaelEischer) [inline cmd/restic/cmd_key.go:145]: Let's change the message a bit: `saved new key with ID %s`

File: changelog/unreleased/issue-4656 [hunk 2]
```diff
@@ -0,0 +1,9 @@
+Bugfix: Properly report the ID of newly added keys
+
+When adding a key to a repository, we had no "easy" way to know the new key's ID.
```
Comment (MichaelEischer) [inline changelog/unreleased/issue-4656:3]: Please shorten this to something similar to the following:
```
`restic key add` now reports the ID of a newly added key. This simplifies
selecting a specific key using the `--key-hint keyID` option.
```

**Review by MichaelEischer — CHANGES_REQUESTED** [review]
The ideal solution would be to add JSON support to `key add`. But I agree that the current output is not useful. So I'm fine with changing it, I just just have a few small remarks as commented below.

**Comment by ls-initiatives** [comment]
Updates following your suggestions, thanks.
I agree that JSON output would be nice to have, though I can't write the patch for it. I'll file an issue about that if you want.

**Comment by MichaelEischer** [comment]
> I agree that JSON output would be nice to have, though I can't write the patch for it. I'll file an issue about that if you want.

That would be great.

**Review by MichaelEischer — APPROVED** [review]
LGTM.
